server/midddleware/forwarder.go

```diff
@@ -0,0 +1,88 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/mwitkow/grpc-proxy/proxy"
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/types"
+	"github.com/tigrisdata/tigris/util"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+var (
+	UserAgent = "tigris-forwarder/" + util.Version
+)
+
+var clients map[string]*grpc.ClientConn
+
+// TODO: Sweep unused connections periodically
+func getClient(ctx context.Context, origin string) (*grpc.ClientConn, error) {
+	if c, ok := clients[origin]; ok {
+		return c, nil
+	}
+
+	opts := []grpc.DialOption{
+		grpc.FailOnNonTempDialError(true),
+		grpc.WithReturnConnectionError(),
+		grpc.WithUserAgent(UserAgent),
+		grpc.WithBlock(),
+	}
+
+	conn, err := grpc.DialContext(ctx, origin, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	clients[origin] = conn
+
+	return conn, nil
+}
+
+func proxyDirector(ctx context.Context, _ string) (context.Context, *grpc.ClientConn, error) {
+	txCtx := api.GetTransaction(ctx, nil)
+
+	client, err := getClient(ctx, txCtx.GetOrigin())
```

Comment:
return `ctx, getClient(ctx, txCtx.GetOrigin())`

Reply:
It doesn't look work this way.